### ISSUE TITLE
[Issue]: rsmi_dev_energy_count_get total power measurement

### ISSUE THREAD

**Issue by breyerml** [RadeonOpenCompute/rocm_smi_lib#180]
### Problem Description

The ROCm SMI function `rsmi_dev_energy_count_get` should return the total energy consumed since the last driver restart. However, the output is constant over time during one program execution. Between two program executions, the total energy changes. 

Example: 
first execution:
```
5198685762204
5198685762204
5198685762204
5198685762204
5198685762204
5198685762204
5198685762204
5198685762204
5198685762204
5198685762204
```

second execution:
```
5198710150404
5198710150404
5198710150404
5198710150404
5198710150404
5198710150404
5198710150404
5198710150404
5198710150404
5198710150404
```

My question: is it possible to get the real total energy consumption during one program execution and not only a constant value?

### Operating System

Ubuntu 22.04.4 LTS (Jammy Jellyfish)

### CPU

AMD EPYC 9274F 24-Core Processor

### GPU

AMD Instinct MI210

### ROCm Version

ROCm 6.1.0

### ROCm Component

rocm_smi_lib

### Steps to Reproduce

Using the CMakeLists.txt:
```CMake
cmake_minimum_required(VERSION 3.1...3.29)

project(Test VERSION 1.0 LANGUAGES CXX)

add_executable(prog main.cpp)

find_package(rocm_smi REQUIRED)
target_link_libraries(prog PRIVATE -lrocm_smi64)
target_include_directories(prog PRIVATE ${ROCM_SMI_INCLUDE_DIR})
```
and the main.cpp:
```c++
#include <iostream>
#include <thread>
#include <chrono>
#include <atomic>

#include "rocm_smi/rocm_smi.h"

using namespace std::chrono_literals;

std::atomic<bool> finished{ false };

void loop() {
  rsmi_init(0);

  while (!finished) {
    [[maybe_unused]] std::uint64_t timestamp{};
    [[maybe_unused]] float resolution{};

    std::uint64_t power{};
    rsmi_dev_energy_count_get(0, &power, &resolution, &timestamp);
    std::cout << power << std::endl;

    std::this_thread::sleep_for(500ms);
  }

  rsmi_shut_down();
}

int main() {
  std::thread t{ [&]() { loop(); } };

  std::this_thread::sleep_for(5s);
  
  finished = true;
  t.join();

  return 0;
}
```
the output can be reproduced using `mkdir build && cd build && cmake .. && ./prog`.

Although no GPU kernels are running, the GPU should draw at least some energy. Additionally, it also does not work in our real code that in fact does use the GPU for multiple seconds.

### (Optional for Linux users) Output of /opt/rocm/bin/rocminfo --support

```
OCk module is loaded
=====================    
HSA System Attributes    
=====================    
Runtime Version:         1.13
Runtime Ext Version:     1.4
System Timestamp Freq.:  1000.000000MHz
Sig. Max Wait Duration:  18446744073709551615 (0xFFFFFFFFFFFFFFFF) (timestamp count)
Machine Model:           LARGE                              
System Endianness:       LITTLE                             
Mwaitx:                  DISABLED
DMAbuf Support:          NO

==========               
HSA Agents               
==========               
*******                  
Agent 1                  
*******                  
  Name:                    AMD EPYC 9274F 24-Core Processor   
  Uuid:                    CPU-XX                             
  Marketing Name:          AMD EPYC 9274F 24-Core Processor   
  Vendor Name:             CPU                                
  Feature:                 None specified                     
  Profile:                 FULL_PROFILE                       
  Float Round Mode:        NEAR                               
  Max Queue Number:        0(0x0)                             
  Queue Min Size:          0(0x0)                             
  Queue Max Size:          0(0x0)                             
  Queue Type:              MULTI                              
  Node:                    0                                  
  Device Type:             CPU                                
  Cache Info:              
    L1:                      32768(0x8000) KB                   
  Chip ID:                 0(0x0)                             
  ASIC Revision:           0(0x0)                             
  Cacheline Size:          64(0x40)                           
  Max Clock Freq. (MHz):   4050                               
  BDFID:                   0                                  
  Internal Node ID:        0                                  
  Compute Unit:            48                                 
  SIMDs per CU:            0                                  
  Shader Engines:          0                                  
  Shader Arrs. per Eng.:   0                                  
  WatchPts on Addr. Ranges:1                                  
  Features:                None
  Pool Info:               
    Pool 1                   
      Segment:                 GLOBAL; FLAGS: FINE GRAINED        
      Size:                    197804460(0xbca41ac) KB            
      Allocatable:             TRUE                               
      Alloc Granule:           4KB                                
      Alloc Recommended Granule:4KB                                
      Alloc Alignment:         4KB                                
      Accessible by all:       TRUE                               
    Pool 2                   
      Segment:                 GLOBAL; FLAGS: KERNARG, FINE GRAINED
      Size:                    197804460(0xbca41ac) KB            
      Allocatable:             TRUE                               
      Alloc Granule:           4KB                                
      Alloc Recommended Granule:4KB                                
      Alloc Alignment:         4KB                                
      Accessible by all:       TRUE                               
    Pool 3                   
      Segment:                 GLOBAL; FLAGS: COARSE GRAINED      
      Size:                    197804460(0xbca41ac) KB            
      Allocatable:             TRUE                               
      Alloc Granule:           4KB                                
      Alloc Recommended Granule:4KB                                
      Alloc Alignment:         4KB                                
      Accessible by all:       TRUE                               
  ISA Info:                
*******                  
Agent 2                  
*******                  
  Name:                    AMD EPYC 9274F 24-Core Processor   
  Uuid:                    CPU-XX                             
  Marketing Name:          AMD EPYC 9274F 24-Core Processor   
  Vendor Name:             CPU                                
  Feature:                 None specified                     
  Profile:                 FULL_PROFILE                       
  Float Round Mode:        NEAR                               
  Max Queue Number:        0(0x0)                             
  Queue Min Size:          0(0x0)                             
  Queue Max Size:          0(0x0)                             
  Queue Type:              MULTI                              
  Node:                    1                                  
  Device Type:             CPU                                
  Cache Info:              
    L1:                      32768(0x8000) KB                   
  Chip ID:                 0(0x0)                             
  ASIC Revision:           0(0x0)                             
  Cacheline Size:          64(0x40)                           
  Max Clock Freq. (MHz):   4050                               
  BDFID:                   0                                  
  Internal Node ID:        1                                  
  Compute Unit:            48                                 
  SIMDs per CU:            0                                  
  Shader Engines:          0                                  
  Shader Arrs. per Eng.:   0                                  
  WatchPts on Addr. Ranges:1                                  
  Features:                None
  Pool Info:               
    Pool 1                   
      Segment:                 GLOBAL; FLAGS: FINE GRAINED        
      Size:                    198117388(0xbcf080c) KB            
      Allocatable:             TRUE                               
      Alloc Granule:           4KB                                
      Alloc Recommended Granule:4KB                                
      Alloc Alignment:         4KB                                
      Accessible by all:       TRUE                               
    Pool 2                   
      Segment:                 GLOBAL; FLAGS: KERNARG, FINE GRAINED
      Size:                    198117388(0xbcf080c) KB            
      Allocatable:             TRUE                               
      Alloc Granule:           4KB                                
      Alloc Recommended Granule:4KB                                
      Alloc Alignment:         4KB                                
      Accessible by all:       TRUE                               
    Pool 3                   
      Segment:                 GLOBAL; FLAGS: COARSE GRAINED      
      Size:                    198117388(0xbcf080c) KB            
      Allocatable:             TRUE                               
      Alloc Granule:           4KB                                
      Alloc Recommended Granule:4KB                                
      Alloc Alignment:         4KB                                
      Accessible by all:       TRUE                               
  ISA Info:                
*******                  
Agent 3                  
*******                  
  Name:                    gfx90a                             
  Uuid:                    GPU-76651bd391bcae4c               
  Marketing Name:          AMD Instinct MI210                 
  Vendor Name:             AMD                                
  Feature:                 KERNEL_DISPATCH                    
  Profile:                 BASE_PROFILE                       
  Float Round Mode:        NEAR                               
  Max Queue Number:        128(0x80)                          
  Queue Min Size:          64(0x40)                           
  Queue Max Size:          131072(0x20000)                    
  Queue Type:              MULTI                              
  Node:                    2                                  
  Device Type:             GPU                                
  Cache Info:              
    L1:                      16(0x10) KB                        
    L2:                      8192(0x2000) KB                    
  Chip ID:                 29711(0x740f)                      
  ASIC Revision:           1(0x1)                             
  Cacheline Size:          64(0x40)                           
  Max Clock Freq. (MHz):   1700                               
  BDFID:                   8960                               
  Internal Node ID:        2                                  
  Compute Unit:            104                                
  SIMDs per CU:            4                                  
  Shader Engines:          8                                  
  Shader Arrs. per Eng.:   1                                  
  WatchPts on Addr. Ranges:4                                  
  Coherent Host Access:    FALSE                              
  Features:                KERNEL_DISPATCH 
  Fast F16 Operation:      TRUE                               
  Wavefront Size:          64(0x40)                           
  Workgroup Max Size:      1024(0x400)                        
  Workgroup Max Size per Dimension:
    x                        1024(0x400)                        
    y                        1024(0x400)                        
    z                        1024(0x400)                        
  Max Waves Per CU:        32(0x20)                           
  Max Work-item Per CU:    2048(0x800)                        
  Grid Max Size:           4294967295(0xffffffff)             
  Grid Max Size per Dimension:
    x                        4294967295(0xffffffff)             
    y                        4294967295(0xffffffff)             
    z                        4294967295(0xffffffff)             
  Max fbarriers/Workgrp:   32                                 
  Packet Processor uCode:: 82                                 
  SDMA engine uCode::      8                                  
  IOMMU Support::          None                               
  Pool Info:               
    Pool 1                   
      Segment:                 GLOBAL; FLAGS: COARSE GRAINED      
      Size:                    67092480(0x3ffc000) KB             
      Allocatable:             TRUE                               
      Alloc Granule:           4KB                                
      Alloc Recommended Granule:2048KB                             
      Alloc Alignment:         4KB                                
      Accessible by all:       FALSE                              
    Pool 2                   
      Segment:                 GLOBAL; FLAGS: EXTENDED FINE GRAINED
      Size:                    67092480(0x3ffc000) KB             
      Allocatable:             TRUE                               
      Alloc Granule:           4KB                                
      Alloc Recommended Granule:2048KB                             
      Alloc Alignment:         4KB                                
      Accessible by all:       FALSE                              
    Pool 3                   
      Segment:                 GROUP                              
      Size:                    64(0x40) KB                        
      Allocatable:             FALSE                              
      Alloc Granule:           0KB                                
      Alloc Recommended Granule:0KB                                
      Alloc Alignment:         0KB                                
      Accessible by all:       FALSE                              
  ISA Info:                
    ISA 1                    
      Name:                    amdgcn-amd-amdhsa--gfx90a:sramecc+:xnack-
      Machine Models:          HSA_MACHINE_MODEL_LARGE            
      Profiles:                HSA_PROFILE_BASE                   
      Default Rounding Mode:   NEAR                               
      Default Rounding Mode:   NEAR                               
      Fast f16:                TRUE                               
      Workgroup Max Size:      1024(0x400)                        
      Workgroup Max Size per Dimension:
        x                        1024(0x400)                        
        y                        1024(0x400)                        
        z                        1024(0x400)                        
      Grid Max Size:           4294967295(0xffffffff)             
      Grid Max Size per Dimension:
        x                        4294967295(0xffffffff)             
        y                        4294967295(0xffffffff)             
        z                        4294967295(0xffffffff)             
      FBarrier Max Size:       32                                 
*** Done *** 
```

### Additional Information

The program output with `export RSMI_LOGGING=1`: [debug_output.txt](https://github.com/user-attachments/files/15807966/debug_output.txt)

**Comment by ppanchad-amd** [comment]
@breyerml Internal ticket has been created to assist with your issue. Thanks!

**Comment by jamesxu2** [comment]
Hello @breyerml, 

Thanks for providing your sample program. Can you update ROCm to the recently released 6.2 version (i.e. using [amdgpu-installer method](https://rocm.docs.amd.com/projects/install-on-linux/en/latest/install/amdgpu-install.html) or any other) and try again? 

I do see the cumulative energy counter incrementing between calls to **rsmi_dev_energy_ct_get** within a single program execution on my system, running an updated version of ROCm with MI210s. I also annotated the delta between power readings and it seems to hover between the two values obtained.

```
$ ./main
8789699213541 
8789700637715 delta=1.04858e+06
8789702050512 delta=1.57286e+06
8789703470408 delta=1.57286e+06
8789704888501 delta=1.57286e+06
8789706310418 delta=1.04858e+06
8789707729883 delta=1.57286e+06
8789709151249 delta=1.57286e+06
8789710567155 delta=1.04858e+06
8789711987485 delta=1.57286e+06
```

**Comment by breyerml** [comment]
Hello @jamesxu2,

sorry for the late reply (but I was on vacation). 
Yes, updating to ROCm 6.2 solves the issue. Thanks a lot!
